scipy/stats/_multivariate.py

```diff
@@ -5739,3 +5741,292 @@ def _sample_uniform_direction(dim, size, random_state):
     samples = random_state.standard_normal(samples_shape)
     samples /= np.linalg.norm(samples, axis=-1, keepdims=True)
     return samples
+
+
+_dirichlet_mn_doc_default_callparams = """\
+alpha : 1D array_like
+    The concentration parameters. The number of entries determines the
+    dimensionality of the distribution. Each entry must be strictly positive.
+n : int
+    The number of trials. Must be a strictly positive integer (scalar); this
+    distribution is not vectorized to support multiple shape parameters.
+"""
+
+_dirichlet_mn_doc_frozen_callparams = ""
+
+_dirichlet_mn_doc_frozen_callparams_note = """\
+See class definition for a detailed description of parameters."""
+
+dirichlet_mn_docdict_params = {
+    '_dirichlet_mn_doc_default_callparams': _dirichlet_mn_doc_default_callparams,  # noqa
+    '_doc_random_state': _doc_random_state
+}
+
+dirichlet_mn_docdict_noparams = {
+    '_dirichlet_mn_doc_default_callparams': _dirichlet_mn_doc_frozen_callparams,
+    '_doc_random_state': _doc_random_state
+}
+
+def _dirichlet_multinomial_check_parameters(alpha, n, x=None):
+
+    alpha = np.asarray(alpha)
+    if alpha.ndim != 1:
+        raise ValueError("`alpha` must be an array with exactly one dimension.")
+
+    n = np.asarray(n)
+    if n.ndim != 0:
+        raise ValueError("`n` must be an integer (scalar).")
+
+    if x is not None:
+        # Ensure that `x` and `alpha` are arrays. If the shapes are
+        # incompatible, NumPy will raise an appropriate error.
+        try:
+            x, alpha = np.broadcast_arrays(x, alpha)
+        except ValueError as e:
+            msg = "`x` and `alpha` must be broadcastable."
+            raise ValueError(msg) from e
+
+        x_int = np.floor(x)
+        if np.any(x < 0) or np.any(x != x_int):
+            raise ValueError("`x` must contain only non-negative integers.")
+        x = x_int
+
+    if np.any(alpha <= 0):
+        raise ValueError("`alpha` must contain only positive values.")
+
+    n_int = np.floor(n)
+    if np.any(n <= 0) or np.any(n != n_int):
+        raise ValueError("`n` must be a positive integer.")
```

Comment:
Since we don't expect `n` to be an array, we can simplify this as:

```suggestion
    n_int = np.floor(n)
    if n <= 0 or n != n_int:
        raise ValueError("`n` must be a positive integer.")
```

This will have the nice side-effect of erroring out when `n` is non-scalar. I think we should also check beforehand if `n` is a scalar or not and raise an exception with a clear message that the distribution is not vectorized.

Reply:
I just noticed that we are checking on line 5776 whether `n` is a scalar or not. Then I think the above suggestion becomes optional and for readability only.

Reply:
Thanks. We're going to add back the ability for `n` to be an array in a separate PR, so shall we leave it alone?

Reply:
Sure!